parse.js

```diff
@@ -609,6 +621,47 @@ var parse = function (buf) {
     })
   })
 
+  schema.messages.forEach(function (msg) {
+    msg.fields.forEach(function (field) {
+      if (field.options && field.options.packed === 'true') {
+        if (PACKABLE_TYPES.indexOf(field.type) === -1) {
+          // let's see if it's an enum
+          if (field.type.indexOf('.') === -1) {
+            if (msg.enums && msg.enums.some(function (en) {
+              return en.name === field.type
+            })) {
+              return
+            }
+          } else {
+            var fieldSplit = field.type.split('.')
+            if (fieldSplit.length > 2) {
+              throw new Error('what is this?')
```

Comment:
oh yes, sorry, that one slipped out.
To summarize (maybe I should add a comment...): a packed field must be a repeated field of primitive numeric type. There is a fixed list + enums. To figure out whether the type is an enum, I know of 2 cases:

1. the enum is declared in the message type (in `valid-packed.proto`, see `Corpus` type in `ValidPacked` message)
1. the enum is declared in a different message and reused via the "dot notation" (`EnumCarrying.E`)

So I split the field type at the `.`s. I expect the resulting array to have a length of 1 or 2. `>2` would be erroneous. I'm not sure what is the right thing to do here.
Maybe that's the right occasion for a [`console.assert`](https://nodejs.org/api/console.html#console_console_assert_value_message)?